portability-spi-transfer/src/main/java/org/dataportabilityproject/spi/transfer/provider/ExportResult.java

```diff
@@ -0,0 +1,109 @@
+package org.dataportabilityproject.spi.transfer.provider;
+
+import org.dataportabilityproject.spi.transfer.types.ContinuationData;
+import org.dataportabilityproject.types.transfer.models.DataModel;
+
+/**
+ * The result of an item export operation.
+ */
+public class ExportResult<T extends DataModel> {
+    public static final ExportResult CONTINUE = new ExportResult(ResultType.CONTINUE);
+    public static final ExportResult END = new ExportResult(ResultType.CONTINUE);
+
+    /**
+     * Result types.
+     */
+    public enum ResultType {
+        /**
+         * Indicates the operation was successful and more items are available so the export should continue.
+         */
+        CONTINUE,
+        /**
+         * Indicates the operation was successful and no more items are available.
+         */
+        END,
+        /**
+         * Indicates an unrecoverable error was raised.
+         */
+        ERROR,
+
+        /**
+         * Indicates a recoverable error was raised.
+         */
+        RETRY
+    }
+
+    private ResultType type;
```

Comment:
No necessary right now but one idea that comes to mind is that with multiple constructors, this may be a good candidate for the Builder pattern or static factory method pattern, then we can make these final and set them in a private constructor.